docs/user-guide/projects.md

```diff
@@ -54,6 +54,75 @@ argocd proj add-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 argocd proj remove-destination <PROJECT> <CLUSTER>,<NAMESPACE>
 ```
 
+We can also do negations of destinations (i.e. install anywhere _apart from_).
+
+```bash
+argocd proj add-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+argocd proj remove-destination <PROJECT> !<CLUSTER>,!<NAMESPACE>
+```
+
+Declaratively we could do something like this:
+
+```yaml
+spec:
+  destinations:
+  # Do not allow any app to be installed in `kube-system`  
+  - namespace: '!kube-system'
+    server: '*'
+  # Or any cluster that has a URL of `team1-*`   
+  - namespace: '*'
+    server: '!https://team1-*'
+    # Any other namespace or server is fine though.
+  - namespace: '*'
+    server: '*'
+```
+
+Doing this is order-dependent though, and this could also lead to unexpected behavior. Take this example:
+
+```yaml
+spec:
+  destinations:
+  # Allow any app to be installed in `kube-system`  
+  - namespace: 'kube-system'
+    server: '*'
+  # Or not?
+  - namespace: '!kube-system'
+    server: '*'
+```
+
+This would mean that any app can be allowed to be installed into `kube-system`, which is _probably_ not what you wanted.
+To get around this we have a field on the project spec called `allDestinationsMustMatch`, which means that all the 
+`destination` rules must be `true` for a given destination. 
+
+```yaml
+spec:
+  # With this field in place, nothing can be installed in `kube-system`.
+  allDestinationsMustMatch: true
+  destinations:
+  - namespace: 'kube-system'
```

Comment:
if we have allDestinationsMustMatch = false , what about precedence of rule applied?

Reply:
I'm not sure if I'm misunderstanding what you're asking for, but AFAIK if `allDestinationsMustMatch = false`, the precedence depends on the order of `destinations` itself.

Reply:
I don't think there's any "precedence" in that case exactly. I think we just loop over destinations and return `true` on the first one that matches. Otherwise return `false`. So order is insignificant.

Reply:
what about when we add two rules from cli ?

argocd proj add-destination <PROJECT> !<CLUSTER>,!<NAMESPACE1>
argocd proj add-destination  <PROJECT> !<CLUSTER>,!<NAMESPACE2>

there will be no way to have allDestinationsMustMatch , and even if we give option for this via cli , there will be precedence issue right? it will be unclear that which got applied